api/src/oscc.c

```diff
@@ -107,29 +107,35 @@ oscc_result_t oscc_open( unsigned int channel )
 
 oscc_result_t oscc_close( unsigned int channel )
 {
-    oscc_result_t result = OSCC_ERROR;
+    oscc_result_t ret = OSCC_ERROR;
 
-    if( global_oscc_can_socket < 0 )
+    if( global_oscc_can_socket >= 0 )
     {
         int result = close( global_oscc_can_socket );
 
-        if ( result > 0 )
+        if ( result == 0 )
         {
-            result = OSCC_OK;
+            ret = OSCC_OK;
         }
     }
 
-    if( global_vehicle_can_socket < 0 )
+    if( global_vehicle_can_socket >= 0 )
     {
         int result = close( global_vehicle_can_socket );
 
-        if ( result > 0 )
+        if ( result == 0 && ret == OSCC_OK )
```

Comment:
Why assign `ret = OSCC_OK` here if a condition of getting there is that `ret == OSCC_OK`?

Reply:
Originally I decided to take a simpler approach in this code and see if anyone cared in the code review, but I'm not sure that this is correct. The tricky bit is that the previous code has 3 different code paths and this was a simple, if crude approach - but now that I think about it this isn't correct. If for whatever reason the OSCC CAN socket wasn't opened `ret` will remain at the default value of `OSCC_ERROR`, which is unlikely to happen but probably isn't desirable.

Part of the problem is what should happen if both sockets are already closed - should that be an error? Is the return code of this function indicating whether an action was taken or if the sockets are closed?

Reply:
So the documentation is as follows:

```c
/**
 * @brief Use provided CAN channel to close communications
 *        to CAN bus connected to the OSCC modules.
 *
 * @param [in] channel - CAN channel connected to OSCC modules.
 *
 * @return OSCC_ERROR or OSCC_OK
 *
 */
```

Based on this, I think the most reasonable thing to do is error if both channels are closed, error if either channel failed to close, and return ok if a channel was closed and no errors occurred. I'll push this as a separate commit that we can yank and redo if another approach is chosen.

Reply:
@shnewto I've done the mentioned refactor. Could you take a look?